src/main/java/com/aws/greengrass/deployment/DeploymentDocumentDownloader.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment;
+
+import com.amazon.aws.iot.greengrass.configuration.common.Configuration;
+import com.aws.greengrass.deployment.converter.DeploymentDocumentConverter;
+import com.aws.greengrass.deployment.exceptions.DeploymentDocumentDownloadException;
+import com.aws.greengrass.deployment.model.DeploymentDocument;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.network.HttpClientProvider;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.Digest;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import com.aws.greengrass.util.SerializerFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import software.amazon.awssdk.services.greengrassv2data.model.GetDeploymentConfigurationRequest;
+import software.amazon.awssdk.services.greengrassv2data.model.GetDeploymentConfigurationResponse;
+import software.amazon.awssdk.utils.IoUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.security.NoSuchAlgorithmException;
+import javax.inject.Inject;
+
+public class DeploymentDocumentDownloader {
+    private static final Logger logger = LogManager.getLogger(DeploymentDocumentDownloader.class);
+
+    private final GreengrassV2DataClient greengrassV2DataClient;
+    private final HttpClientProvider httpClientProvider;
+
+    private String thingName;   // not final as it could be dynamically changed
+
+
+    /**
+     * Constructor.
+     *
+     * @param greengrassServiceClientFactory greengrassServiceClientFactory
+     * @param deviceConfiguration            deviceConfiguration
+     * @param httpClientProvider             httpClientProvider
+     */
+    @Inject
+    public DeploymentDocumentDownloader(GreengrassServiceClientFactory greengrassServiceClientFactory,
+                                        DeviceConfiguration deviceConfiguration,
+                                        HttpClientProvider httpClientProvider) {
+        this.greengrassV2DataClient = greengrassServiceClientFactory.getGreengrassV2DataClient();
```

Comment:
Don't create the client here. Create it as needed.

Reply:
I thought that if `DeploymentDocumentDownloader` is loaded, then this client should be created. But thinking twice, it makes sense - it doesn't have to be created even in the constructor. Changed it to when invoking.